### ISSUE TITLE
Wrong result when integrating Legendre Polynomial (missing case distinction)

### ISSUE THREAD

**Issue by terben** [sympy/sympy#27298]
I ask SymPy to evaluate the following integral:

$$
\int_{-1}^{1} P_l(x) dx,
$$

where $P_l$ is the Legendre polynomial of order $l$. The integral should yield 2 for $l=0$ and 0 for $l>0$. However, SymPy gives only back the result of 0 when I integrate with $l$ being an undetermined integer:

```
import sympy as sp
# SymPy version is 1.13.2

x = sp.symbols('x', real=True)
l = sp.symbols('l', integer=True)

# The following just yields a result of zero (no cases)
sp.Integral(sp.legendre(l, x), (x, -1, 1)).doit().simplify()

# The result for the polynomial of degree 0 is correct 
# when I explicitely ask for it:
sp.Integral(sp.legendre(0, x), (x, -1, 1)).doit().
```

**Comment by oscarbenjamin** [comment]
This seems to be a relevant bug:
```python
In [80]: legendre(-1, -1)
Out[80]: 1

In [81]: legendre(y-1, -1).subs({y: 0})
Out[81]: -1
```
I guess this formula is wrong:
```python
In [82]: legendre(y-1, -1)
Out[82]: 
    y - 1
(-1) 
```
I think it comes from here:
https://github.com/sympy/sympy/blob/a7719e719c0b43ec1dbb964b01b57c4f3783be8d/sympy/functions/special/polynomials.py#L830-L835

**Comment by oscarbenjamin** [comment]
This diff appears to fix the observed bugs:
```diff
diff --git a/sympy/functions/special/polynomials.py b/sympy/functions/special/polynomials.py
index 5816baef60..c45ad64fb7 100644
--- a/sympy/functions/special/polynomials.py
+++ b/sympy/functions/special/polynomials.py
@@ -831,7 +831,7 @@ def eval(cls, n, x):
         if not n.is_Number:
             # Symbolic result L_n(x)
             # L_n(-x)  --->  (-1)**n * L_n(x)
-            if x.could_extract_minus_sign():
+            if x.could_extract_minus_sign() and n.is_nonnegative:
                 return S.NegativeOne**n * legendre(n, -x)
             # L_{-n}(x)  --->  L_{n-1}(x)
             if n.could_extract_minus_sign() and not(-n - 1).could_extract_minus_sign():
```

**Comment by solvedbiscuit71** [comment]
@oscarbenjamin I would like to work on this issue.

**Comment by solvedbiscuit71** [comment]
Hi @terben, If I understood your question correctly, you expect the result of the Integration to be a piecewise function like

```python
Piecewise((2, Eq(l, 0)), (0, l>0))
```

Am I right?

**Comment by terben** [comment]
> Hi [@terben](https://github.com/terben), If I understood your question correctly, you expect the result of the Integration to be a piecewise function like
> 
> Piecewise((2, Eq(l, 0)), (0, l>0))
> 
> But the fix [@oscarbenjamin](https://github.com/oscarbenjamin) proposed does solve this issue by resulting an expression which can be substituted later
> 
> expr = (legendre(l - 1, -1) - legendre(l + 1, -1))/(2*l + 1)
> expr.subs({l: 0}) # returns 2
> expr.subs({l: 1}) # returns 0

Yes, the piecewise solution was my initial expectation and it certainly would be the most useful result. But any solution which does produce a more complicated but correct result is better than the current state.

I have no experience with the SymPy sources and how involved it would be to implement the optimal solution. Perhaps we can implement @oscarbenjamin proposal for a quick fix but leave the *optimal solution* as a feature request?

**Comment by solvedbiscuit71** [comment]
Yes I agree, I have put a PR with a quick fix

**Comment by oscarbenjamin** [comment]
My suggested code change above is not correct/sufficient. I just show it to highlight where in the codebase the bug is so that someone else can investigate how to fix it.

What is needed here is to verify if the formulae that are applied for `legendre` are correct. Currently on master `legendre` applies this transformation using the code I highlighted:
```python
In [1]: legendre(y, -x)
Out[1]: 
    y               
(-1) ⋅legendre(y, x)
```
What is needed is to identify the conditions under which this transformation holds and to check for those conditions before applying the transformation. I don't know what the conditions are but this formula seems to be out by a minus sign for negative integer values of `y`.

When observing the conditions it is important to understand that the arguments to Legendre can both be arbitrary complex numbers as in the Legendre function: the bug here likely stems from thinking only of Legendre polynomials i.e. by assuming that `y` is a nonnegative integer. We need to compare this formula for different values of `y` and `x` e.g. with mpmath:
```python
In [6]: import mpmath

In [7]: mpmath.legendre(1.2, 1.2)
Out[7]: mpf('1.2680749336921637')

In [8]: mpmath.legendre(1.2, -1.2)
Out[8]: mpc(real='-0.88809187691788383', imag='-0.74535574482600953')

In [9]: mpmath.legendre(1.2, 1.2)*(-1)**1.2
Out[9]: mpc(real='-1.0258941714978451', imag='-0.74535574482600941')
```
Once that evaluation rule is fixed the next question is whether or not this antiderivative is correct for all possible values of x and y:
```python
In [10]: integrate(legendre(y, x), x)
Out[10]: 
-legendre(y - 1, x) + legendre(y + 1, x)
────────────────────────────────────────
                2⋅y + 1 
```
Again I suspect that this assumes that `y` is an integer or at least that it is not equal to `-1/2`.

> I have no experience with the SymPy sources and how involved it would be to implement the optimal solution.

What is needed here is someone with the ability to verify the formulae involving Legendre functions or to determine what conditions are required in each case.

**Comment by terben** [comment]
> What is needed here is someone with the ability to verify the formulae involving Legendre functions or to determine what conditions are required in each case.

@oscarbenjamin Thanks a lot for your comments and for indicating where to start. I will look into this asap.
